File: Python/baskara.py
import math

def bhaskara():
    a = float(input("Digite o coeficiente 'a': "))
    b = float(input("Digite o coeficiente 'b': "))
    c = float(input("Digite o coeficiente 'c': "))
    
    delta = b**2 - 4*a*c
    if delta < 0:
        return "Sem raízes reais"
    elif delta == 0:
        x = -b / (2*a)
        return "Raiz única:", x
    else:
        x1 = (-b + math.sqrt(delta)) / (2*a)
        x2 = (-b - math.sqrt(delta)) / (2*a)
        return "Raízes:", x1, "e", x2

# Exemplo de uso
print(bhaskara())

File: C/baskara.c
#include <stdio.h>
#include <math.h>

int main()
{
    double a, b, c, delta, x1, x2;

    printf("coeficiente a: ");
    scanf("%lf", &a);
    printf("coeficiente b: ");
    scanf("%lf", &b);
    printf("coeficiente c: ");
    scanf("%lf", &c);

    delta = b * b - 4 * a * c;

    if (a == 0 || delta < 0)
    {
        printf("Esta equação não possui raízes reais\n");
    }
    else
    {
        x1 = (-b + sqrt(delta)) / (2 * a);
        x2 = (-b - sqrt(delta)) / (2 * a);
        printf("X1 = %.4lf\n", x1);
        printf("X2 = %.4lf\n", x2);
    }

    return 0;
}
